dependency/catalog_services.go

```diff
@@ -71,11 +71,21 @@ func (d *CatalogServicesQuery) Fetch(clients *ClientSet, opts *QueryOptions) (in
 	default:
 	}
 
-	opts = opts.Merge(&QueryOptions{
+	// this overrides the query params present in the query
+	// i.e. overrides the namespace and partition params with ones used the first time
+	// while creating NewCatalogServicesQuery
+
+	// see bug [https://github.com/hashicorp/consul-template/pull/1842#issuecomment-1915723565]
+	// it should be other way around.
+	// default to the query params present while creating NewCatalogServicesQuery
+	// and then merge with the query params present in the query
```

Comment:
Should we have this comment here as the updated code will not have reference of what was used earlier?

Reply:
yes, some of the other functions have that problem too.
that's why I want to enable ENT and write proper tests for them before touching them.